lib/AST/ASTImporter.cpp

```diff
@@ -2899,6 +2899,20 @@ Decl *ASTNodeImporter::VisitFieldDecl(FieldDecl *D) {
       if (Importer.IsStructurallyEquivalent(D->getType(),
                                             FoundField->getType())) {
         Importer.MapImported(D, FoundField);
+        // In case of a FieldDecl of a ClassTemplateSpecializationDecl, the
+        // initializer of a FieldDecl might not had been instantiated in the
+        // "To" context.  However, the "From" context might instantiated that,
+        // thus we have to merge that.
+        if (Expr *FromInitializer = D->getInClassInitializer()) {
+          // We don't have yet the initializer set.
+          if (FoundField->hasInClassInitializer() &&
+              !FoundField->getInClassInitializer()) {
+            Expr *ToInitializer = Importer.Import(FromInitializer);
+            if (!ToInitializer)
+              return nullptr;
```

Comment:
Probably it is not good to return `nullptr` here. `MapImported` was already called, so the new field is seen as imported. The `FoundField` can be returned, or ignore the errors, or call `MapImported` only if every import is successful and return `nullptr` if not.

Reply:
Yeah, this is a good point :+1: 
I think, the only valid option then to return with the `FoundField`, even if we cannot import the Initializer.

The other options would not work, becasue if we call `MapImported` after `Import(FromInitializer)` then we end up having an infinite loop, because `VisitCXXDefaultInitExpr` starts with importing the `FieldDecl`.